zebra-network/src/config.rs

```diff
@@ -46,10 +49,29 @@ impl Config {
     }
 
     /// Get the initial seed peers based on the configured network.
-    pub fn initial_peers(&self) -> HashSet<SocketAddr> {
+    pub async fn initial_peers(&self) -> HashSet<SocketAddr> {
         match self.network {
-            Network::Mainnet => Config::parse_peers(self.initial_mainnet_peers.clone()),
-            Network::Testnet => Config::parse_peers(self.initial_testnet_peers.clone()),
+            Network::Mainnet => Config::parse_peers(self.initial_mainnet_peers.clone()).await,
+            Network::Testnet => Config::parse_peers(self.initial_testnet_peers.clone()).await,
+        }
+    }
+
+    async fn resolve(s: String) -> Option<impl Iterator<Item = SocketAddr>> {
+        // Resolve using `lookup_host`
+        let fut = tokio::net::lookup_host(s.clone());
+        // Add timeout
+        let fut = tokio::time::timeout(Duration::from_nanos(1), fut);
+
+        match fut.await {
+            Ok(Ok(ips)) => Some(ips),
+            Ok(Err(e)) => {
+                tracing::warn!("Can't resolve {:?}. Error: {:?}", s, e);
+                None
```

Comment:
You might need to take a generic type. And it might be easier to convert to a HashSet before returning.

Let's also be a bit clearer with the function and argument names, and document the function.

Since we've documented the function, we don't really need comments inside the function as well. We use comments inside functions to explain how complicated code works, or why code is written in a particular way.

(Sorry about the overlapping suggestions, I thought it would be easier to read them separately.)

```suggestion
    /// Resolves `host` into zero or more IP addresses.
    ///
    /// If `host` is a DNS name, performs DNS resolution with a timeout of a few
    seconds. If DNS resolution fails or times out, returns an empty list.
    async fn resolve_host<S: ToSocketAddrs>(host: S) -> HashSet<SocketAddr> {
        let fut = tokio::net::lookup_host(host);
        let fut = tokio::time::timeout(DNS_LOOKUP_TIMEOUT, fut);

        match fut.await {
            Ok(Ok(ips)) => ips.collect(),
            Ok(Err(e)) => {
                tracing::info!(?host, ?e, "DNS error resolving peer IP address");
                HashSet::new()
```

Reply:
Agree on everything except for the generic type argument. I dont see a need to do that yet, i might change my mind later as i see some evidence on why it is better than the string. By now i added all the other suggestions at https://github.com/ZcashFoundation/zebra/pull/1662/commits/8a0c173391102153bee50edfc1ee90e3fa69179e

Reply:
If we don't need the generic, we can just leave it out.